<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>530</width>
    <height>600</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Поиск кратчайшего пути в городской дорожной цепи</string>
  </property>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>20</y>
     <width>141</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Путь до графа:</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="GraphLineEdit">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>20</y>
     <width>311</width>
     <height>20</height>
    </rect>
   </property>
   <property name="readOnly">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QPushButton" name="GraphButton">
   <property name="geometry">
    <rect>
     <x>480</x>
     <y>20</y>
     <width>41</width>
     <height>21</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="cursor">
    <cursorShape>PointingHandCursor</cursorShape>
   </property>
   <property name="toolTip">
    <string>Выберите файл с описанием графа</string>
   </property>
   <property name="text">
    <string>...</string>
   </property>
  </widget>
  <widget class="QGroupBox" name="StartPopulationGroupBox">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>50</y>
     <width>251</width>
     <height>71</height>
    </rect>
   </property>
   <property name="title">
    <string>Способ формирования начальной популяции</string>
   </property>
   <widget class="QRadioButton" name="EliteRadioButton">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>30</y>
      <width>110</width>
      <height>20</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>PointingHandCursor</cursorShape>
    </property>
    <property name="text">
     <string>Элитный</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QRadioButton" name="FullRadioButton">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>30</y>
      <width>110</width>
      <height>20</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>PointingHandCursor</cursorShape>
    </property>
    <property name="text">
     <string>Полный</string>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="GeneticAlgoGroupBox">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>50</y>
     <width>251</width>
     <height>71</height>
    </rect>
   </property>
   <property name="title">
    <string>Тип генетического алгоритма</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
   <widget class="QRadioButton" name="ClassicRadioButton">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>30</y>
      <width>110</width>
      <height>20</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>PointingHandCursor</cursorShape>
    </property>
    <property name="text">
     <string>Классический</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QRadioButton" name="GenitorRadioButton">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>30</y>
      <width>110</width>
      <height>20</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>PointingHandCursor</cursorShape>
    </property>
    <property name="text">
     <string>Генитор</string>
    </property>
   </widget>
  </widget>
  <widget class="QPushButton" name="FindButton">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>440</y>
     <width>251</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>13</pointsize>
    </font>
   </property>
   <property name="cursor">
    <cursorShape>PointingHandCursor</cursorShape>
   </property>
   <property name="toolTip">
    <string>Поиск кратчайшего пути</string>
   </property>
   <property name="text">
    <string>Найти путь</string>
   </property>
  </widget>
  <widget class="QPushButton" name="SaveButton">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>440</y>
     <width>251</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>13</pointsize>
    </font>
   </property>
   <property name="cursor">
    <cursorShape>PointingHandCursor</cursorShape>
   </property>
   <property name="toolTip">
    <string>Сохранение результатов о кратчайшем пути</string>
   </property>
   <property name="text">
    <string>Сохранить результаты</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="ResultsLineEdit">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>520</y>
     <width>361</width>
     <height>20</height>
    </rect>
   </property>
   <property name="toolTip">
    <string>Кратчайший путь</string>
   </property>
   <property name="readOnly">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="Line" name="line">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>490</y>
     <width>511</width>
     <height>20</height>
    </rect>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Plain</enum>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>520</y>
     <width>141</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Наилучшая хромосома:</string>
   </property>
  </widget>
  <widget class="QGroupBox" name="groupBox">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>230</y>
     <width>511</width>
     <height>191</height>
    </rect>
   </property>
   <property name="title">
    <string>Настройка числовых параметров генетического алгоритма</string>
   </property>
   <widget class="QSpinBox" name="MutationSpinBox">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>110</y>
      <width>71</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="value">
     <number>1</number>
    </property>
   </widget>
   <widget class="QLabel" name="MutationLineEdit">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>110</y>
      <width>401</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Вероятность мутации в хромосоме (%):</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="TournamentSpinBox">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>150</y>
      <width>71</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="value">
     <number>100</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>150</y>
      <width>401</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Вероятность выбора лучшей хромосомы в турнирной селекции (%):</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>401</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Количество хромосом в начальной популяции:</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="HromosomeSpinBox">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>30</y>
      <width>71</width>
      <height>20</height>
     </rect>
    </property>
    <property name="maximum">
     <number>9999</number>
    </property>
    <property name="value">
     <number>20</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>70</y>
      <width>401</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Число итераций генетического алгоритма:</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="IterationSpinBox">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>70</y>
      <width>71</width>
      <height>20</height>
     </rect>
    </property>
    <property name="maximum">
     <number>9999</number>
    </property>
    <property name="value">
     <number>20</number>
    </property>
   </widget>
  </widget>
  <widget class="QLabel" name="label_6">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>560</y>
     <width>150</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Время пути:</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>560</y>
     <width>150</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Длина пути:</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="TimeLineEdit">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>560</y>
     <width>100</width>
     <height>20</height>
    </rect>
   </property>
   <property name="readOnly">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QLineEdit" name="LengthLineEdit">
   <property name="geometry">
    <rect>
     <x>420</x>
     <y>560</y>
     <width>100</width>
     <height>20</height>
    </rect>
   </property>
   <property name="readOnly">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QGroupBox" name="groupBox_2">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>140</y>
     <width>511</width>
     <height>71</height>
    </rect>
   </property>
   <property name="title">
    <string>Тип мутации</string>
   </property>
   <widget class="QRadioButton" name="OnePointRadioButton">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>30</y>
      <width>150</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Одноточечная</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QRadioButton" name="ManyPointsRadioButton">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>30</y>
      <width>150</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Многоточечная</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="TotalRadioButton">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>30</y>
      <width>150</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Тотальная</string>
    </property>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
